cmd/client/command/object.go

```diff
@@ -96,23 +96,33 @@ func updateObjectCmd() *cobra.Command {
 }
 
 func deleteObjectCmd() *cobra.Command {
+	var specFile string
+	var allFlag bool
 	cmd := &cobra.Command{
-		Use:     "delete",
-		Short:   "Delete an object",
-		Example: "egctl object delete <object_name>",
-		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return errors.New("requires one object name to be deleted")
+		Use:   "delete",
+		Short: "Delete an object from a yaml file or name",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if allFlag {
+				handleRequest(http.MethodDelete, makeURL(objectsURL), nil, cmd)
+				return
 			}
 
-			return nil
-		},
+			if len(specFile) != 0 {
+				visitor := buildSpecVisitor(specFile, cmd)
+				visitor.Visit(func(s *spec) error {
+					handleRequest(http.MethodDelete, makeURL(objectURL, s.Name), nil, cmd)
+					return nil
+				})
+				visitor.Close()
+				return
+			}
 
-		Run: func(cmd *cobra.Command, args []string) {
 			handleRequest(http.MethodDelete, makeURL(objectURL, args[0]), nil, cmd)
```

Comment:
args[0] could be invalid after this change.
and `delete` is different from `create` or `update` as it only requires a name instead of full YAML spec, so it is much more easier for the user to make a fault operation. and we should do more checks for the arguments, like what to do if there're both `--all` and `spec file` in the command line?